api/capi/include/nntrainer.h

```diff
@@ -65,9 +72,31 @@ typedef enum {
   ML_LAYER_TYPE_UNKNOWN    /**< Unknown Lyaer */
 } ml_layer_type_e;
 
+/**
+ * @brief Dataset generator callback function for train/valid/test data.
+ * @details The Containers passed will already be allocated with sufficient
+ * space to contain the data by the caller. This function should return a batch
+ * of data in the passed containers.
+ * @note This function can be called multiple times in parallel.
+ * @param[out] input Container to provide input data to the model.
+ * @param[out] label Container to provide label data to the model to train/test.
+ * @param[out] last Container to provide is the data is finished.
```

Comment:
I feel hard to catch what this does from the description

Reply:
Updated description.

Reply:
Nice :) Just one minor thing.
```suggestion
 * @param[out] last Container to provide is the data finished.
```